sae_lens/sae_training_runner.py

```diff
@@ -187,58 +185,56 @@ def _init_sae_group_b_decs(
     def save_checkpoint(
         self,
         trainer: SAETrainer,
-        checkpoint_name: int | str,
+        checkpoint_name: str,
         wandb_aliases: list[str] | None = None,
-    ) -> str:
-
-        checkpoint_path = f"{trainer.cfg.checkpoint_path}/{checkpoint_name}"
-
-        os.makedirs(checkpoint_path, exist_ok=True)
+    ) -> None:
+        base_path = Path(self.cfg.checkpoint_path) / checkpoint_name
+        base_path.mkdir(exist_ok=True, parents=True)
 
-        path = f"{checkpoint_path}"
-        os.makedirs(path, exist_ok=True)
+        self.activations_store.save(str(base_path / "activation_store_state.json"))
 
+        # comment:
+        # Is this broken? it seems like this is "assymetrical" in the sense that
+        # this changes the output of the SAE.
+        # Should we not also balance this by scaling the encoder weights?
         if self.sae.cfg.normalize_sae_decoder:
             self.sae.set_decoder_norm_to_unit_norm()
```

Comment:
I probably am missing something, it seems like this is "asymmetrical" in the sense that it changes the decoder without accordingly scaling the other SAE params. Should we not also balance this by scaling the other encoder weights?

Reply:
ah interesting. A little off topic but could this optimizer instability be fixed by also scaling the other network parameters when the decoder norm is forced to 1? I guess there may be other optimizer instability introduced by scaling everything but should be better?

Reply:
either way, are you suggesting we remove it here?

Reply:
This has been fixed by normalizing the L1 penalty by the decoder norm (`cfg.scale_sparsity_penalty_by_decoder_norm`). This was from one of the Anthropic updates. This removes the pressure on the SAE to shrink the encoder / grow the decoder to reduce L1.

We should not delete this, since it's possible the decoder norm could be slightly larger than 1 without this here.